micrometer-core/src/test/java/io/micrometer/core/instrument/dropwizard/MicrometerMetricRegistryListenerTest.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2019 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.dropwizard;
+
+import com.codahale.metrics.*;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MicrometerMetricRegistryListenerTest {
+
+    private static final String METRIC_PREFIX = "prefix";
+    private static final String METRIC_NAME = "metric";
+    private MeterRegistry meterRegistry = new SimpleMeterRegistry();
```

Comment:
If the `MeterRegistry` is the dropwizard flavor, will we get into a loop, or will it recover gracefully?

Reply:
If you use the same MetricRegistry in both, yes it will loop. But why would you do that ?
(Note that the JMX and Graphite autoconf create their own MetricRegistry)

Reply:
Darn I was hoping it would just hit it twice, but 'add' wouldn't get called for subsequent times. (Just as a safety net).

You wouldn't intentionally do that. But it would be good to test it or note it in the javadoc so users can avoid getting burned.

Are there any autoconfigs that wire up a DropWizard registry? That magic could surprise someone if a registry was created and added to the composite without the user realizing.

Reply:
The system loops because there's a prefix that is added to the metric name. I've pushed a commit to not add a prefix by default. This is however not enough since now you'll get another exception `throw new IllegalArgumentException("A metric named " + name + " already exists");`.
I don't see how to completely prevent this but also I don't think this can really happen in real life...